src/services/executionEnvironment.ts

```diff
@@ -0,0 +1,316 @@
+            } else {
+              this.connection.console.error(
+                `Exception in ExecutionEnvironment service: ${JSON.stringify(error)}`
+              );
+            }
+        }
+
+    }
+
+    async fetchPluginDocs(): Promise<void> {
+      const ansibleConfig = await this.context.ansibleConfig;
+      const containerName = `${this._container_image.replace(/[^a-z0-9]/ig, '_')}`
+
+      try {
+        const containerImageIdCommand = `${this._container_engine} images ${this._container_image} --format="{{.ID}}" | head -n 1`
+        this.connection.console.log(containerImageIdCommand)
+        this._container_image_id = child_process.execSync(containerImageIdCommand, {
+          encoding: 'utf-8'
+        }).trim();
+        const hostCacheBasePath = path.resolve(`${process.env.HOME}/.cache/ansible-language-server/${containerName}/${this._container_image_id}`)
+
+        if (!this.runContainer(containerName)) {
+          return
+        }
+
+        if (fs.existsSync(hostCacheBasePath)) {
+          ansibleConfig.collections_paths = this.updateCachePaths(ansibleConfig.collections_paths, hostCacheBasePath)
+          ansibleConfig.module_locations = this.updateCachePaths(ansibleConfig.module_locations, hostCacheBasePath)
+        } else {
+          ansibleConfig.collections_paths = await this.copyPluginDocFiles(hostCacheBasePath,
+                                  containerName,
+                                  ansibleConfig.collections_paths,
+                                  '**/ansible_collections')
+
+          const builtin_plugin_locations: string[] = []
+          ansibleConfig.module_locations.forEach(modulePath => {
+            const pluginsPathParts = modulePath.split(path.sep).slice(0, -1)
+            if (modulePath.includes(`${path.sep}site-packages${path.sep}`)) {
+              pluginsPathParts.push('plugins')
+            }
+            builtin_plugin_locations.push(pluginsPathParts.join(path.sep))
+          })
+          // Copy builtin plugins
+          await this.copyPluginDocFiles(hostCacheBasePath,
+            containerName,
+            builtin_plugin_locations,
+            '*')
+
+          // Copy builtin modules
+          ansibleConfig.module_locations = await this.copyPluginDocFiles(hostCacheBasePath,
+            containerName,
+            ansibleConfig.module_locations,
+            '**/modules')
+        }
+      } catch(error) {
+        this.connection.window.showErrorMessage(
+          `Exception in ExecutionEnvironment service while fetching docs: ${JSON.stringify(error)}`
+        )
+      } finally {
+        this.cleanUpContainer(containerName)
+      }
+    }
+
+    public wrapContainerArgs(command: string): string {
+      const workspaceFolderPath = URI.parse(this.context.workspaceFolder.uri).path
+      const containerCommand: Array<string> = [this._container_engine]
+      containerCommand.push(...['run', '--rm', '--interactive'])
```

Comment:
why do we need `--interactive`?

Reply:
Removed `--interactive` flag as it is not required for our use case.
 It is required if we have to pass user inputs to ansible command for example privilege escalation password on `ansible-playbook` command execution which is not required in this case. I had referred `ansible-runner` logic for container command wrap which does support executing `ansible-playbook` within EE.

Reply:
fixed